### PR TITLE
Replaced catch-all  with a custom listener. Fixes #1

### LINKED ISSUE #1: Don't use robot.hear /.*/
Awesome work!

"match everything" listeners (`robot.hear /.*/`) are an anti-pattern and should be replaced with middleware. In this case, I _think_ [receive middleware](https://hubot.github.com/docs/scripting/#receive-middleware) is the right solution, but if not, let's talk about what the ideal interface would look like, and I'll get it built into the core.

Filing this now so I don't forget; I'll do some more noodling soon and give more detailed suggestions.

### PR DESCRIPTION
**Pull request by lmarkus**
Replaces Anti-pattern mentioned in #1

### CODE REVIEW

**Comment by lmarkus** [comment]
What do you think @michaelansel ?

File: lib/index.js [hunk 1]
```diff
@@ -5,17 +5,22 @@
...
 module.exports = function Conversation(bot) {
 
-    var _talkingTo = {},//TODO: Use the robot brain for this, it will probably be more scalable.
-        spy = function (msg) {
+    var _talkingTo = {};//TODO: Use the robot brain for this, it will probably be more scalable.
+
+    //Register a custom listener that will spy on all incoming messages
+    bot.listen(
+        function matcher(msg) {
+            return true; //All messages are accepted.
```
Comment (michaelansel) [inline lib/index.js]: This is still going to have the same behavior (and problems) as `robot.hear /.*/`, buuut what if you instead only return true when the message matches one of the choices? That would behave properly (matcher returns true iff the message is going to get processed).

Comment (lmarkus) [inline lib/index.js]: Ah... Gotcha.
I thought the issue was specific to `hear`.
Simple enough.

**Comment by lmarkus** [comment]
@michaelansel Implemented the change, now it will only accept a message if it belongs to an ongoing conversation.

I decided not to check for matches against the current choice set, given that I may implement some default handler soon.

My expectation is that once a dialog is open (or continued), the next message is part of that conversation as long as it comes from the next user.

**Comment by michaelansel** [comment]
Looks good to me! I can't think of any situations where that will cause major problems.

**Comment by lmarkus** [comment]
Thx! Appreciate the review + tips

**Pull request merged**